pysteps/nowcasts/steps.py

```diff
@@ -33,6 +36,971 @@
     DASK_IMPORTED = False
 
 
+@dataclass
+class StepsNowcasterConfig:
```

Comment:
Since pysteps dropped support for python 3.9:
`Optional[float]` can be written as `float | None`
`Dict[str, Any]` can be written as `dict[str, Any]` and same goes for `List`

Doing this saves some imports from the typing module; you don't have to import `Optional`, `Dict` and `List`